skt/__init__.py

```diff
@@ -152,27 +151,35 @@ def __init__(self, uri, ref=None, wdir=None):
         self.info = []
         self.mergelog = "%s/merge.log" % self.wdir
 
-        try:
+        # Create the work directory if it does not exist
+        if not os.path.isdir(self.wdir):
             os.mkdir(self.wdir)
-        except OSError:
-            pass
 
-        try:
+        # Remove the old merge log
+        if os.path.isfile(self.mergelog):
             os.unlink(self.mergelog)
-        except OSError:
-            pass
 
-        self.git_cmd("init")
+        # Set the origin remote for the repository
+        self.initialize_repo()
+        self.set_repo_remote()
 
+        logging.info("base repo url: %s", self.uri)
+        logging.info("base ref: %s", self.ref)
+        logging.info("work dir: %s", self.wdir)
+
+    def initialize_repo(self):
+        """Initialize the git repository"""
+        git_directory = "{}/.git".format(self.wdir)
+        if not os.path.isdir(git_directory):
+            self.git_cmd("init")
+
+    def set_repo_remote(self):
+        """Set the origin remote for the repository"""
```

Comment:
I appreciate that you're trying to make the `__init__` function easier to read. However, I think that benefit here is marginal, if not negative. Each function (`initialize_repo` and `set_repo_remote`) is just a few lines long and line-count overhead is bigger than the functions amount to. The code in functions depends on certain variables being initialized already, and that's not obvious at the place of their invocation, which can lead to issues after code is reordered. That could be fixed by making them accept the values in question as arguments, but then they're so short, that would add significant syntax overhead. Better effect can be achieved by just adding comments to the original code, I think. Lastly, I don't know what exact conventions in Python are, but I would make the functions private, if I were adding them.

Do you think you can leave the code in place and just add comments explaining what it does?